tcs/csp_solver_mono_eq_methods.cpp

```diff
@@ -28,1882 +28,960 @@ OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+    {
+        m_t_ts_calc = t_ts_cr_su;       //[s]
+    }
+    else if (m_pc_mode == C_csp_power_cycle::STARTUP || m_pc_mode == C_csp_power_cycle::STARTUP_CONTROLLED)
+    {
+        m_t_ts_calc = t_ts_pc_su;       //[s]
+    }
+    
+    if (m_solver_mode == E__TO_PC_PLUS_TES_FULL__ITER_M_DOT_SU || m_solver_mode == E__CR_OUT__ITER_M_DOT_SU
+        || m_solver_mode == E__TO_PC__ITER_M_DOT_SU)
+    {
+        *diff_target = (m_dot_tes_guess - mpc_csp_solver->mc_pc_out_solver.m_m_dot_htf) / mpc_csp_solver->mc_pc_out_solver.m_m_dot_htf;
+    }
+    else if (m_solver_mode == E__CR_OUT__ITER_Q_DOT_TARGET)
+    {
+        *diff_target = (mpc_csp_solver->mc_pc_out_solver.m_q_dot_htf - m_q_dot_pc_target) / m_q_dot_pc_target;
+    }
+
+    return 0;
 }
 
-int C_csp_solver::C_MEQ_cr_df__pc_off__tes_full__T_cold::operator()(double T_htf_cold /*C*/, double *diff_T_htf_cold /*-*/)
+void C_csp_solver::C_MEQ__T_field_cold::init_calc_member_vars()
 {
-	mpc_csp_solver->mc_cr_htf_state_in.m_temp = T_htf_cold;		//[C]
-
-	mpc_csp_solver->mc_collector_receiver.on(mpc_csp_solver->mc_weather.ms_outputs,
-		mpc_csp_solver->mc_cr_htf_state_in,
-		m_defocus,
-		mpc_csp_solver->mc_cr_out_solver,
-		mpc_csp_solver->mc_kernel.mc_sim_info);
-
-	// Check if receiver is OFF or model didn't solve
-	if (mpc_csp_solver->mc_cr_out_solver.m_m_dot_salt_tot == 0.0 || mpc_csp_solver->mc_cr_out_solver.m_q_thermal == 0.0)
-	{
-		*diff_T_htf_cold = std::numeric_limits<double>::quiet_NaN();
-		return -1;
-	}
-
-	// Get receiver HTF outlet temperature
-	double T_htf_rec_hot = mpc_csp_solver->mc_cr_out_solver.m_T_salt_hot;		//[C]
-
-	// Solve TES for *full* charge
-	double T_htf_tes_cold, m_dot_tes;
-	T_htf_tes_cold = m_dot_tes = std::numeric_limits<double>::quiet_NaN();
-	mpc_csp_solver->mc_tes.charge_full(mpc_csp_solver->mc_kernel.mc_sim_info.ms_ts.m_step,
-		mpc_csp_solver->mc_weather.ms_outputs.m_tdry + 273.15,
-		T_htf_rec_hot + 273.15,
-		T_htf_tes_cold,
-		m_dot_tes,
-		mpc_csp_solver->mc_tes_outputs);
-
-	T_htf_tes_cold -= 273.15;	//[C] convert from K
-	m_dot_tes *= 3600.0;		//[kg/hr] convert from kg/s
-
-	// HTF charging state
-	mpc_csp_solver->mc_tes_ch_htf_state.m_m_dot = m_dot_tes;				//[kg/hr]
-	mpc_csp_solver->mc_tes_ch_htf_state.m_temp_in = mpc_csp_solver->mc_cr_out_solver.m_T_salt_hot;		//[C]
-	mpc_csp_solver->mc_tes_ch_htf_state.m_temp_out = T_htf_tes_cold;		//[C]
-
-	// If not actually discharging (i.e. mass flow rate = 0.0), what should the temperatures be?
-	mpc_csp_solver->mc_tes_dc_htf_state.m_m_dot = 0.0;										//[kg/hr]
-	mpc_csp_solver->mc_tes_dc_htf_state.m_temp_in = mpc_csp_solver->mc_tes_outputs.m_T_cold_ave - 273.15;	//[C] convert from K
-	mpc_csp_solver->mc_tes_dc_htf_state.m_temp_out = mpc_csp_solver->mc_tes_outputs.m_T_hot_ave - 273.15;	//[C] convert from K
-
-	*diff_T_htf_cold = (T_htf_tes_cold - T_htf_cold) / T_htf_cold;	//[-]
-	
-	return 0;
+    m_t_ts_calc = std::numeric_limits<double>::quiet_NaN();
```

Comment:
also rename this "m_t_ts_calc" and all similar variables

Reply:
current name is ok